src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,386 @@
+                var issue = await client.Issue.Get(repo.Id, issueNumber.Value);
+                var descriptionUpdateFlag = false;
+                var comments = await client.Issue.Comment.GetAllForIssue(repo.Id, issue.Number);
+                var updateIssueDescription = CreateOrUpdateDescription(updateHistoryError, issue.Body);
+                // check if the issue body has to be updated or comment body has to be updated.
+                if (updateIssueDescription.Item2)
+                {
+                    // Issue body will be updated
+                    descriptionUpdateFlag = true;
+                    var issueUpdate = issue.ToUpdate();
+                    var bodyTitle = $"The following errors have been detected when attempting to update dependencies in " +
+                        $"'{updateHistoryError.Repository}' \n\n";
+                    issueUpdate.Body = bodyTitle + updateIssueDescription.Item1 + "\n\n**/FyiHandle :** " + _options.FyiHandle;
+                    issueUpdate.State = ItemState.Open;
+                    Logger.LogInformation($"Updating issue body for the issue :  '{issueNumber.Value}'");
+                    await client.Issue.Update(repo.Id, issueNumber.Value, issueUpdate);
+                }
+                foreach (var comment in comments)
+                {
+                    descriptionUpdateFlag = false;
+                    var updateCommentDescription = CreateOrUpdateDescription(updateHistoryError, comment.Body);
+                    if (updateCommentDescription.Item2)
+                    {
+                        // Issue's comment body will be updated
+                        descriptionUpdateFlag = true;
+                        Logger.LogInformation($"Updating comment body for the issue : '{issueNumber.Value}'");
+                        await client.Issue.Comment.Update(repo.Id, comment.Id, updateCommentDescription.Item1);
+                        continue;
+                    }
+                }
+                // New error for the same repo-branch combination, so adding a new comment to it. 
+                if (!descriptionUpdateFlag && !string.IsNullOrEmpty(updateIssueDescription.Item1))
+                {
+                    Logger.LogInformation($"Creating a new comment for the issue : '{issueNumber.Value}'");
+                    await client.Issue.Comment.Create(repo.Id, issueNumber.Value, updateIssueDescription.Item1);
+                }
+            }
+            else
+            {
+                // Create a new issue for the error
+                var labelName = "DependencyUpdateError";
+                Logger.LogInformation(
+                    $"Dependency Update Error, for the error message  '{updateHistoryError.ErrorMessage}'" +
+                    $"for the repository : '{updateHistoryError.Repository}'");
+                var createIssue = new NewIssue("[Dependency Update] Errors during dependency updates to :" + updateHistoryError.Repository);
```

Comment:
I think you mean createdIssue as the name of this variable.

Reply:
You're using string concatenation here.

Reply:
changed it